### ISSUE TITLE
Inheritance is not checked inside prototype object literals

### ISSUE THREAD

**Issue by crdev** [google/closure-compiler#707]
Consider the following case:

```
        /** @constructor */
        var MyType = function() {};
        MyType.prototype = {
          /** @return {number} */
          method: function() {}
        }
        /**
         * @constructor 
         * @extends {MyType}
         */
        var MyOtherType = function() {};
        MyOtherType.prototype = {
          /**
           * @return {string}
           * @override
           */
          method: function() { return ''; }
        }
```

The override's return type is not the same as that of the superclass member. Yet the compiler does not complain about this. The issue is that the TypeCheck pass does no inheritance checks inside the prototype object literals.
